### DISCUSSION TITLE
Are controls (blanks) removed in the output?

### DISCUSSION THREAD

**Discussion by NeMaVo** [Shenhav-and-Korem-labs/SCRuB#21] [Q&A] [answered]
Hello,
I am trying to use an iterative approach of using SCRuB because I have multiple types of controls (blanks). In my first iteration, the output decontaminated_samples matrix does not contain the controls anymore. So when I continue on my iteration using this output as input for the next, I loose these controls to track well-to-well leakage at a later stage. 
Is that expected behaviour (aka the function generally does not output controls), or were they maybe removed because their ASV count was 0 after decontamination?

Thank you!

**Comment by gaustin15** [comment, marked as answer]
Hi, thank you for the post. 

To the question – yes, it is expected behavior that the negative controls are not included in the `decontaminated_samples` output matrix. 

It is accurate to say that under the SCRuB model, negative controls which are both de-leaked and decontaminated are going to have zero reads (this is because under this model, nothing other than those two components can be present in the controls). If you are looking into using the inferred content of the controls that is contamination during downstream analyses, the output's `gamma` component is a vector that represents the relative abundance of scrub’s estimated contamination source.

However, to the point about using these negative controls to track for well-to-well leakage at a later stage-- an underlying assumption behind scrub (and I believe this statement would hold for most decontamination methods) is that any ‘well-to-well leakage’ for microbes originating in a negative control should be treated as contamination. As a result, scrub only models well-to-well leakage from nearby samples into the negative controls (and the other direction is captured in the model as contamination). Because of this, in general I don’t think there is a need to use these controls to track for well-to-well leakage in later stages, since the SCRuB model has already removed those contaminants from the samples during its decontamination step. 

Please let me know if I'm misinterpreting the use case for which you want to use these controls to track for well-to-well leakage, or you have any additional questions.

**Reply by NeMaVo** [reply]
Hello again and thank you for the quick response!

Your answer makes sense, thanks! 

I do have one more question (and there might be more coming). Please let me know if I should open another thread for this.

Is there any way to find out how much of an ASV got removed because of leakage vs. because of "external" contamination? For example, when I check which ASVs have the highest gamma values as suggested here [#2 ](https://github.com/Shenhav-and-Korem-labs/SCRuB/issues/2), and compare with the ASVs which got completely removed in most samples, they do not match. My guess was that the gamma value is depicting both the contribution as "external" contamination and "internal" leakage together. So I wonder if I can somehow disentangle the external from internal contaminants? With the aim being the identification of ASVs that came from my samples and leaked into other samples (or the controls, in the strict sense of SCRuB), and ASVs that come from an external source.

Edit 13.12.2024:
I made this plot, showing ASVs on the x axis and the proportion of reads removed for each ASV on the y axis (1 being 100% of reads of this ASV got removed) across all samples. Is it reasonable to assume that all ASVs that were completely removed from the dataset were "external" contaminants, while those ASVs for which partial reads were removed (>0 & <1) are genuine ASVs that leaked into other samples?
![image](https://github.com/user-attachments/assets/721f1679-fdb3-4c5e-b11f-bde7d844483f)


Thanks again!

**Comment by gaustin15** [comment]
> Is there any way to find out how much of an ASV got removed because of leakage vs. because of "external" contamination? 

In general, classifying an ASV as a binary ‘removed’ or ‘not removed’ (or a 'leaked ASV' vs 'not a leaked ASV') from the contamination source due to leakage deviates from SCRuB’s underlying assumptions, because scrub is modeling leakage on a per-sample level(similarly to the ‘contaminant’ vs ‘not contaminant’ classification of ASVs discussion is the issue you referenced. 

To the ‘ASVs got removed due to leakage' point – since SCRuB is looking into leakage from samples into negative controls, the ASVs that make up the composition of what is estimated to have leaked into controls will bias SCRuB towards keeping those ASVs in the actual samples, since these are removed from the estimated contamination source. So, if we’re looking within the decontaminated samples, no ASVs (nor any proportion of ASVs) are removed due to leakage.


> For example, when I check which ASVs have the highest gamma values as suggested here [#2 ](https://github.com/Shenhav-and-Korem-labs/SCRuB/issues/2), and compare with the ASVs which got completely removed in most samples, they do not match. My guess was that the gamma value is depicting both the contribution as "external" contamination and "internal" leakage together. 

During my answer, I am assuming that by ‘internal contamination’, we’re referring to what scrub denotes as ‘well-to-well leakage’. Please let me know if I’m misinterpreting the use of the ‘internal/external contamination’ terms here.

> So I wonder if I can somehow disentangle the external from internal contaminants? With the aim being the identification of ASVs that came from my samples and leaked into other samples (or the controls, in the strict sense of SCRuB), and ASVs that come from an external source.

If we’re looking for the composition of what scrub has estimated to have leaked into controls, the `alpha` component of scrub’s output represents what percentage of each observed negative control is estimated to have originated from a nearby sample (so alpha is a `n_controls` x `n_samples + 1` matrix, where each row sums to 1). So, using the decontaminated samples matrix (which represents the inferred ‘true’ composition of the samples) , the compositions of each ‘leakage community’, for each control, can be estimated via matrix multiplication between alpha[,1:n_samples] and the relative abundance of the decontaminated samples.


> Edit 13.12.2024:
I made this plot, showing ASVs on the x axis and the proportion of reads removed for each ASV on the y axis (1 being 100% of reads of this ASV got removed) across all samples. Is it reasonable to assume that all ASVs that were completely removed from the dataset were "external" contaminants, while those ASVs for which partial reads were removed (>0 & <1) are genuine ASVs that leaked into other samples?

This would not be an accurate interpretation of SCRuB’s results. As you mention, SCRuB doesn’t track leakage in between samples, but rather only samples into controls. Therefore, observing partially removed ASVs within is not indicative of leakage between samples. We can track the inferred contamination community via the `gamma` output, and any nonzero abundance within this vector indicates that this corresponding ASV is estimated to be (at least in part) a contaminant. 

I believe that what you're looking (per my understanding -- tracking well-to-well leakage) for is more in line with analyzing the `alpha` component of the output as I described above, although it's worth keeping in mind that the binary 'leaked' vs 'not leaked' inference on an per-ASV level is a deviation from scrub's underlying assumptions.
